presto-main/src/main/java/com/facebook/presto/resourcemanager/DistributedTaskInfoResource.java

```diff
@@ -0,0 +1,83 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourcemanager;
+
+import com.facebook.presto.execution.TaskId;
+import com.facebook.presto.server.BasicQueryInfo;
+import com.facebook.presto.spi.QueryId;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+import java.util.Optional;
+
+import static com.facebook.airlift.http.client.HttpUriBuilder.uriBuilderFrom;
+import static com.facebook.presto.server.security.RoleType.ADMIN;
+import static com.facebook.presto.server.security.RoleType.USER;
+import static java.util.Objects.requireNonNull;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+
+@Path("/v1/taskInfo")
+@RolesAllowed({USER, ADMIN})
+public class DistributedTaskInfoResource
+{
+    private final ResourceManagerClusterStateProvider clusterStateProvider;
+    private final ResourceManagerProxy proxyHelper;
+
+    @Inject
+    public DistributedTaskInfoResource(
+            ResourceManagerClusterStateProvider clusterStateProvider,
+            ResourceManagerProxy proxyHelper)
+    {
+        this.clusterStateProvider = requireNonNull(clusterStateProvider, "clusterStateProvider is null");
+        this.proxyHelper = requireNonNull(proxyHelper, "proxyHelper is null");
+    }
+
+    @GET
+    @Path("{taskId}")
+    public void getTaskInfo(@PathParam("taskId") TaskId taskId,
+            @Context UriInfo uriInfo,
+            @Context HttpServletRequest servletRequest,
+            @Suspended AsyncResponse asyncResponse)
+            throws WebApplicationException
+    {
+        proxyTaskInfoResponse(servletRequest, asyncResponse, uriInfo, taskId);
+    }
+
+    private void proxyTaskInfoResponse(HttpServletRequest servletRequest, AsyncResponse asyncResponse, UriInfo uriInfo, TaskId taskId)
+    {
+        QueryId queryId = taskId.getQueryId();
+        Optional<BasicQueryInfo> queryInfo = clusterStateProvider.getClusterQueries().stream()
+                .filter(query -> query.getQueryId().equals(queryId))
+                .findFirst();
+
+        if (queryInfo.isPresent()) {
+            proxyHelper.performRequest(servletRequest, asyncResponse, uriBuilderFrom(queryInfo.get().getSelf()).replacePath(uriInfo.getPath()).build());
```

Comment:
dumb question : once it reaches RM, don't you want the request to be sent to the correct coordinator.
curious why need a replace path with the RM uri here.

Reply:
The URI in `queryInfo.get().getSelf()` will have the path `/v1/query/{queryId}`, not `/v1/taskInfo/{taskId}`, so we want to replace it.

We can check that this is working by printing the URI with/without the `.replacePath(...)` call:
```
System.out.println(uriBuilderFrom(queryInfo.get().getSelf()).replacePath(uriInfo.getPath()).build());
System.out.println(uriBuilderFrom(queryInfo.get().getSelf()).build());
System.out.println(queryInfo.get().getSelf());
```
We get
```
http://127.0.0.1:50378/v1/taskInfo/20221027_194117_00001_kvd29.1.0.1
http://127.0.0.1:50378/v1/query/20221027_194117_00001_kvd29
http://127.0.0.1:50378/v1/query/20221027_194117_00001_kvd29
```

Reply:
can you please create a method called createTaskInfoUri and move the uri creation to that.
Without that method it is not clear what is going on.

Reply:
Done

Reply:
Can you use createTaskInfoUri here as well.

Reply:
Fixed